GeosCore/dust_mod.F90

```diff
@@ -211,51 +211,36 @@ SUBROUTINE CHEMDUST( Input_Opt,  State_Chm, State_Diag, &
-                                      Spc(APMIDS%id_DSTBIN1+7)%Conc(I,J,1)    &
-                                      +E_DST*0.1144517534356116
-          Spc(APMIDS%id_DSTBIN1+8)%Conc(I,J,1)  =                             &
-                                      Spc(APMIDS%id_DSTBIN1+8)%Conc(I,J,1)    &
-                                      +E_DST*0.1880182758312848
-          Spc(APMIDS%id_DSTBIN1+9)%Conc(I,J,1)  =                              &
-                                      Spc(APMIDS%id_DSTBIN1+9)%Conc(I,J,1)    &
-                                      +E_DST*0.2448371224641443
-          Spc(APMIDS%id_DSTBIN1+10)%Conc(I,J,1) =                             &
-                                      Spc(APMIDS%id_DSTBIN1+10)%Conc(I,J,1)   &
-                                      +E_DST*0.1452025570524453
-          Spc(APMIDS%id_DSTBIN1+11)%Conc(I,J,1) =                             &
-                                      Spc(APMIDS%id_DSTBIN1+11)%Conc(I,J,1)   &
-                                      +E_DST*0.1954553759504486
-          Spc(APMIDS%id_DSTBIN1+12)%Conc(I,J,1) =                             &
-                                      Spc(APMIDS%id_DSTBIN1+12)%Conc(I,J,1)   &
-                                      +E_DST*4.0358202390254526d-002
-          Spc(APMIDS%id_DSTBIN1+13)%Conc(I,J,1) =                             &
-                                      Spc(APMIDS%id_DSTBIN1+13)%Conc(I,J,1)   &
-                                      +E_DST*5.2032641469382107d-003
-          Spc(APMIDS%id_DSTBIN1+14)%Conc(I,J,1) =                             &
-                                      Spc(APMIDS%id_DSTBIN1+14)%Conc(I,J,1)   &
-                                      +E_DST*7.2351157743278424d-004
+          Spc(I,J,1,APMIDS%id_DSTBIN1)    = Spc(I,J,1,APMIDS%id_DSTBIN1)    &
+                                            +E_DST*1.5083735055071434d-006
+          Spc(I,J,1,APMIDS%id_DSTBIN1+1)  = Spc(I,J,1,APMIDS%id_DSTBIN1+1)  &
+                                            +E_DST*4.7465319694587000d-005
+          Spc(I,J,1,APMIDS%id_DSTBIN1+2)  = Spc(I,J,1,APMIDS%id_DSTBIN1+2)  &
+                                            +E_DST*6.5459286253814980d-004
+          Spc(I,J,1,APMIDS%id_DSTBIN1+3)  = Spc(I,J,1,APMIDS%id_DSTBIN1+3)  &
+                                            +E_DST*3.6946414378027244d-003
+          Spc(I,J,1,APMIDS%id_DSTBIN1+4)  = Spc(I,J,1,APMIDS%id_DSTBIN1+4)  &
+                                            +E_DST*8.0323357006383631d-003
+          Spc(I,J,1,APMIDS%id_DSTBIN1+5)  = Spc(I,J,1,APMIDS%id_DSTBIN1+5)  &
+                                            +E_DST*1.5922384531190468d-002
+          Spc(I,J,1,APMIDS%id_DSTBIN1+6)  = Spc(I,J,1,APMIDS%id_DSTBIN1+6)  &
+                                            +E_DST*3.7324808601695736d-002
+          Spc(I,J,1,APMIDS%id_DSTBIN1+7)  = Spc(I,J,1,APMIDS%id_DSTBIN1+7)  &
+                                            +E_DST*0.1144517534356116
```

Comment:
Need to use either "d" exponents or "e" exponents with `_fp` to ensure we don't lose precision.